File: two-sum-ii-input-array-is-sorted/two_sum_ii.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
    let left = 0, right = numbers.length-1;
    while(left < right) {
        let s = numbers[left] + numbers[right];
        if(s==target) {
            return [left+1, right+1]
        }else if(s > target) {
            right--;
        }else {
            left++;
        }
    }

    return null;
};

File: two-sum-ii-input-array-is-sorted/two_sum_ii.rb
# @param {Integer[]} numbers
# @param {Integer} target
# @return {Integer[]}
def two_sum(numbers, target)
    left, right = 0, numbers.length-1

    while left < right do
        s = numbers[left] + numbers[right]
        if s == target
            return [left+1, right+1]
        elsif s > target
            right-=1
        else
            left+=1
        end
    end
end